### DISCUSSION TITLE
Repeated Bonvision stimulus using Range Animation does not work when triggered with Subscribe When node

### DISCUSSION THREAD

**Discussion by austenfisher** [bonsai-rx/bonsai#1325] [Q&A] [answered]
Hello,

In previous versions of Bonsai and Bonvision, I was able to produce a visual stimulus that consisted of an expanding disc that repeated 5 times with a small delay between each. The visual stimulus was triggered with the 'key down' and 'subscribe when' nodes. After an update from a couple of months ago, the visual stimulus works as intended for the first expansion of the disc, but all subsequent repetitions display the circle at its starting size as dictated by a 'range animation' node without the expected expansion. If I remove the key down trigger to allow the animation to run as soon as I start the bonsai file, the stimulus displays as intended. 

Is there another way that I can trigger the animation so that it displays as intended?

I've attached a file that displays the animation as intended w/o the key down trigger and another that uses the key down trigger and produces the incorrect animation. The workflow uses a series of grouped nodes as shown below.

![image](https://user-images.githubusercontent.com/87714691/232914921-ddddcb1f-4cc2-4e2c-8684-5521b0461e7b.png)

![image](https://user-images.githubusercontent.com/87714691/232914960-0d32696c-d9a5-4168-b942-431ade38db8f.png)

![image](https://user-images.githubusercontent.com/87714691/232915766-75c2a892-f4d1-424d-bf29-cfe00e8bfa5b.png)

![image](https://user-images.githubusercontent.com/87714691/232915803-51b0c5ce-6a69-4289-bedf-d78646b44f14.png)

Thanks for any help in advance!
[bonvisionRangeAnimation.zip](https://github.com/bonsai-rx/bonsai/files/11266688/bonvisionRangeAnimation.zip)

**Comment by bruno-f-cruz** [comment, marked as answer]
Heya,

You should make the event that controls the end of each observable explicit, otherwise, it will be a really hard job to understand when the "repeat" node is triggered.

See attached a more explicit solution using `SelectMany`.
[loom_SelectMany.zip](https://github.com/bonsai-rx/bonsai/files/11270320/loom_SelectMany.zip)
cheers,
B

**Reply by austenfisher** [reply]
Thanks so much! This works great and I was able to add a one second pause between animations using the same solution.
